elasticgraph-indexer_autoscaler_lambda/lib/elastic_graph/indexer_autoscaler_lambda/concurrency_scaler.rb

```diff
@@ -94,6 +103,22 @@ def get_max_cpu_utilization
         end.max.to_f
       end
 
+      def get_min_free_storage(cluster_name)
+        metric_response = @cloudwatch_client.get_metric_data({
+          start_time: ::Time.now - 1200, # past 20 minutes
+          end_time: ::Time.now,
+          metric_data_queries: [
+            {
+              id: "minFreeStorageAcrossNodes",
+              expression: "SEARCH('{AWS/ES,ClientId,DomainName} MetricName=\"FreeStorageSpace\" AND DomainName=\"#{cluster_name}\"', 'Minimum', 60)",
+              return_data: true
+            }
+          ]
+        })
+
+        metric_response.metric_data_results.first.values.first
```

Comment:
```suggestion
        metric_response.metric_data_results.first.fetch("minFreeStorageAcrossNodes")
```

/nit I'm not sure if this'll work but it's a bit more robust to get the query result by name instead of relying on it being first.

Reply:
It's actually called `values` here https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/CloudWatch/Types/MetricDataResult.html#values-instance_method